Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -312,4 +318,56 @@ extension BrowserViewController: ReaderModeDelegate {
             toolbar.updateReaderModeState(state)
         }
     }
+}
+
+extension BrowserViewController: BrowserLongPressGestureDelegate {
+    func browser(browser: Browser, elementsDidLongPressed elements: [ElementType]) {
+        var actionSheetController = UIAlertController(title: "", message: "Mozilla Firefox", preferredStyle: UIAlertControllerStyle.ActionSheet)
+        var url: NSURL?
+        for element in elements {
+            switch element {
+            case .hrefLink(let urlLink):
```

Comment:
With the dictionary changes mentioned earlier, this function will become a bit nicer. Instead of iterating over the array with the switch statement, you'll need to only do <code>if let linkURL = elements[LongPressElementType.Link] {...}</code> for each.